tests/v2/validation/rbac/rbac_namespace_move_test.go

```diff
@@ -0,0 +1,200 @@
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	//Get cluster name from the config file and append cluster details in rb
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *NameSpaceMoveTestSuite) CreateProjects() {
+	newUser, err := createUser(rb.client, standardUser)
+	require.NoError(rb.T(), err)
+	rb.standardUser = newUser
+	rb.T().Logf("Created user: %v", rb.standardUser.Username)
+	rb.standardUserClient, err = rb.client.AsUser(newUser)
+	require.NoError(rb.T(), err)
+
+	//Create project as a Project Owner
+	createProjectAsPO, err := createProject(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.standardUserPO = createProjectAsPO
+	err = users.AddProjectMember(rb.client, rb.standardUserPO, rb.standardUser, roleProjectOwner)
+	require.NoError(rb.T(), err)
+
+	//Create project as a Project Read Only
+	createProjectAsPR, err := createProject(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.standardUserPR = createProjectAsPR
+	err = users.AddProjectMember(rb.client, rb.standardUserPR, rb.standardUser, roleProjectReadOnly)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *NameSpaceMoveTestSuite) ValidateAdminCanMoveANamespaceWithReadOnlyProject() {
+	// Create namespace under project owner project
+	projectOwnerNamespaceName := namegen.AppendRandomString("testns-")
+	projectOwnerNamespace, err := namespaces.CreateNamespace(rb.client, projectOwnerNamespaceName+"-admin-user-project-owner-permission", "{}", map[string]string{}, map[string]string{}, rb.standardUserPO)
+	require.NoError(rb.T(), err)
+
+	// Create namespace under project read only project
+	projectReadOnlyNamespaceName := namegen.AppendRandomString("testns-")
+	projectReadOnlyNamespace, err := namespaces.CreateNamespace(rb.client, projectReadOnlyNamespaceName+"-admin-user-project-read-only-permission", "{}", map[string]string{}, map[string]string{}, rb.standardUserPR)
+	require.NoError(rb.T(), err)
+
+	// Validate Admin can move namespace from project Owner role to project read only role
+	steveAdminClient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveAdminClient = steveAdminClient
+
+	updateNS, err := getAndConvertNamespace(projectReadOnlyNamespace, rb.steveAdminClient)
+	require.NoError(rb.T(), err)
+	updateNS.Labels = projectOwnerNamespace.Labels
+	updateNS.Annotations = projectOwnerNamespace.Annotations
+
+	_, err = rb.steveAdminClient.SteveType(namespaces.NamespaceSteveType).Update(projectReadOnlyNamespace, updateNS)
+	require.NoError(rb.T(), err)
```

Comment:
Can we add a check to make sure the namespace is indeed moved to the new project? We can may be get the namespace and check the namespace labels which should be the new project labels?